shop/src/main/java/ro/msg/learning/shop/domain/Product.java

```diff
@@ -0,0 +1,38 @@
+package ro.msg.learning.shop.domain;
+
+import jakarta.persistence.*;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.math.BigDecimal;
+
+@Entity
+@Data
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+public class Product extends EntityWithUUID {
+    @Column(name="name")
+    private String name;
+
+    @Column(name="description")
+    private String description;
+
+    @Column(name="price")
+    private BigDecimal price;
+
+    @Column(name="weight")
+    private Double weight;
+
+    @ManyToOne(cascade = {CascadeType.ALL})
+    @JoinColumn(name = "category")
+    private ProductCategory category;
```

Comment:
Cascading is usually recommended and useful for `@OneToMany` mappings. Here, it might lead to unwanted effects. What happens if you have 2 products of the same category and you decide to delete one of the products? 😄 

Reference: https://stackoverflow.com/questions/13027214/what-is-the-meaning-of-the-cascadetype-all-for-a-manytoone-jpa-association (read the top 2 answers)